Momento/ScsDataClient.cs

```diff
@@ -69,11 +71,86 @@ public async Task<CacheSetResponse> SetAsync(string cacheName, string key, byte[
             return new CacheSetResponse(response);
         }
 
-         public async Task<CacheSetResponse> SetAsync(string cacheName, string key, byte[] value)
+        public async Task<CacheSetResponse> SetAsync(string cacheName, string key, byte[] value)
         {
             return await this.SetAsync(cacheName, key, value, defaultTtlSeconds);
         }
 
+        public async Task<CacheMultiGetResponse> MultiGetAsync(string cacheName, List<string> keys)
+        {
+            List<CacheGetResponse> successResponses = new();
+            List<CacheMultiGetFailureResponse> failedResponses = new();
+            foreach (string key in keys)
+            {
+                _GetRequest request = new _GetRequest() { CacheKey = Convert(key) };
+                DateTime deadline = DateTime.UtcNow.AddSeconds(this.dataClientOperationTimeoutSeconds);
+                try
+                {
+                    _GetResponse resp = await this.grpcManager.Client().GetAsync(request, new Metadata { { "cache", cacheName } }, deadline: deadline);
```

Comment:
With how we are currently looping through each key and then awaiting an async call here this will end up being a pretty slow call since its doing them 1 at a time. I think we will want to use async tasks and the WhenAll method for allowing us to fire off all the requests in parallel before awaiting their responses.

https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.whenall?view=netframework-4.7.1